cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
For newly created repositories with enabled garbage collection (`cvmfs_server mkfs -z`) [this config variable defaults to](https://github.com/cvmfs/cvmfs/blob/devel/cvmfs/cvmfs_server#L1668) `3 days ago`. Additionally, I will cover how to configure legacy repositories for garbage collection in a documentation section. This is to be written still...

I don't think that defaulting `CVMFS_GC_TIMESTAMP_THRESHOLD` here is a good idea as there could be repositories that are garbage collectable (`CVMFS_GARBAGE_COLLECTION=true`) but don't want to make use of the automatic garbage collection feature (`CVMFS_GC_TIMESTAMP_THRESHOLD` is set in the configuration).